### PR TITLE
Add cloud-init-vmware-guestinfo package

### PR DESCRIPTION
**Pull request by MateuszMalisz**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR adds cloud-init-vmware-guestinfo package, which is required for the cloud-init to work in the VMWare cloud environment. The package supplies cloud-init datasource for the vSphere's GuestInfo interface.

###### Change Log  

- Added cloud-init-vmware-guestinfo package.

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Local build succeeded.

### CODE REVIEW

File: SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec [hunk 1]
```diff
@@ -0,0 +1,41 @@
+%{!?python3_sitelib: %define python3_sitelib %(python3 -c "from distutils.sysconfig import get_python_lib;print(get_python_lib())")}
+Name:           cloud-init-vmware-guestinfo
+Version:        1.3.1
+Release:        1
+Summary:        A cloud-init datasource for VMware
+Group:          System/Management
+License:        Apache-2.0
```
Comment (thcrain-msft) [inline SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec:7]: Please adjust to match our convention of using Fedora short names for the License field. I believe that would be "ASL 2.0" for this license.

File: SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Url:            https://github.com/vmware/cloud-init-vmware-guestinfo
```
Comment (thcrain-msft) [inline SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec:10]: Consistency nit: about 80% of our current specs use "URL" in all caps for this field.

**Review by thcrain-msft — APPROVED** [review]

File: cgmanifest.json [hunk 3]
```diff
@@ -490,6 +490,17 @@
         }
       }
     },
+    {
+      "component": {
+      "type": "other",
+      "other": {
+        "name": "cloud-init-vmware-guestinfo",
```
Comment (dmcilvaney) [inline cgmanifest.json:497]: Super minor, and will get cleaned up the next time we run a full cgmanifest regeneration, but these should ideally be in sorted order.

**Review by dmcilvaney — APPROVED** [review]

File: SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+install -m 0644 DataSourceVMwareGuestInfo.py %{buildroot}%{python3_sitelib}/cloudinit/sources/DataSourceVMwareGuestInfo.py
+
+%files
+%license LICENSE
+%config %{_sysconfdir}/cloud/cloud.cfg.d/99-DataSourceVMwareGuestInfo.cfg
+%{python3_sitelib}/cloudinit/sources/DataSourceVMwareGuestInfo.py
+
+%changelog
+* Thu Sep 17 2020 Mateusz Malisz <mamalisz@microsoft.com> 1.3.1-1
+- Initial CBL-Mariner packaging.
```
Comment (PawelWMS) [inline SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec:40]: >Initial CBL-Mariner packaging. [](start = 2, length = 30)

For our own specs please use the phrase `Original version for CBL-Mariner.`. See: [point the  21st from our guidelines](https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/125/Introduction-to-SPEC-Files).

File: SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Release:        1
+Summary:        A cloud-init datasource for VMware
+Group:          System/Management
+License:        ASL 2.0
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/vmware/cloud-init-vmware-guestinfo
+BuildArch:      noarch
+
+#Source0:  https://github.com/vmware/%{name}/archive/v%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/cloud-init-vmware-guestinfo/cloud-init-vmware-guestinfo.spec:13]: >   [](start = 9, length = 2)

Nit: could you also align this comment's spacing to other tags and put `BuildArch` behind the sources to align better with other specs?

**Review by PawelWMS — CHANGES_REQUESTED** [review]
:clock1:

File: cgmanifest.json (hunk 3 above)
Comment (dmcilvaney) [reply cgmanifest.json:497]: Disregard, I misread this

**Review by schmittjoseph — APPROVED** [review]

**Review by PawelWMS — APPROVED** [review]
:shipit:

**Pull request merged**